src/pages/dataTableUtilities/reducer/reducerMethods.js

```diff
@@ -447,3 +447,63 @@ export const hideStockOut = state => {
 };
 
 export const showStockOut = state => ({ ...refreshData(state), showAll: true, searchTerm: '' });
+export const editCountedTotalQuantity = (state, action) => {
+  const { rowKey } = action;
+  const { dataState } = state;
+
+  // Change object reference of row in `dataState` to trigger rerender of that row.
+  // Realm object reference in `data` can't be affected in any tidy manner.
+  const newDataState = new Map(dataState);
+  const nextRowState = newDataState.get(rowKey);
+  newDataState.set(rowKey, { ...nextRowState });
+
+  return { ...state, dataState: newDataState };
+};
+
+export const openStocktakeBatchModal = (state, action) => {
+  const { rowKey } = action;
+  const { data } = state;
+
+  const stocktakeBatchToEdit = data.find(({ id }) => id === rowKey);
+
+  return {
+    ...state,
+    modalValue: stocktakeBatchToEdit,
+    modalKey: MODAL_KEYS.EDIT_STOCKTAKE_BATCH,
+  };
+};
+
+export const closeStocktakeBatchModal = state => {
```

Comment:
`refreshData` could probably use `modalKey: ''` - wouldn't have an effect if no modal is open, and wouldn't want to be refreshing the underlying data with a modal open. Has a few use cases